tests/src/core/geometry/testqgspolygon.cpp

```diff
@@ -2854,6 +2854,23 @@ void TestQgsPolygon::toFromWKT()
   pl3.addInteriorRing( compound );
   wkt = pl3.asWkt();
   QCOMPARE( wkt, QStringLiteral( "Polygon ((0 0, 0 10, 10 10, 10 0, 0 0),(1 1, 1 9, 9 9, 9 1, 1 1))" ) );
+
+  // Test WKT export with empty interior ring
+  QgsPolygon pl4;
+  QgsLineString *ext4 = new QgsLineString();
+  ext4->setPoints( QgsPointSequence() << QgsPoint( Qgis::WkbType::Point, 0, 0 )
+                   << QgsPoint( Qgis::WkbType::Point, 0, 10 )
+                   << QgsPoint( Qgis::WkbType::Point, 10, 10 )
+                   << QgsPoint( Qgis::WkbType::Point, 10, 0 )
+                   << QgsPoint( Qgis::WkbType::Point, 0, 0 ) );
+  pl4.setExteriorRing( ext4 );
+  pl4.addInteriorRing( new QgsLineString() );
+  wkt = pl4.asWkt();
+  QCOMPARE( wkt, QStringLiteral( "Polygon ((0 0, 0 10, 10 10, 10 0, 0 0),())" ) );
```

Comment:
This is not a valid WKT

'()' must be removed. The keyword would be EMPTY, but afaik, you could not add an EMPTY ring in a polygon

Reply:
@lbartoletti , I was hoping this PR would trigger a discussion :)

Current state of affairs when creating from WKT in QGIS:

```
>>> poly = QgsGeometry.fromWkt('Polygon ((0 0, 10 10, 0 0),EMPTY)')
>>> poly.asWkt()
'Polygon EMPTY'
```


```
>>> poly = QgsGeometry.fromWkt('Polygon ((0 0, 10 10, 0 0),())')
>>> poly.asWkt()
'Polygon ((0 0, 10 10, 0 0),LineString EMPTY)'
```

```
>>> poly = QgsGeometry.fromWkt('Polygon ((0 0, 10 10, 0 0),LineString EMPTY')
>>> poly.asWkt()
'Polygon EMPTY'
```

This PR insures that we can successfully do: poly.fromWkt(poly.asWkt()). I'm not sure there's a WKT standard for this. But IMHO worth insuring exported strings can be re-used within QGIS.

Reply:
Small state of the art on this subject:

GDAL:
```python
>>> from osgeo import ogr
>>> g = ogr.CreateGeometryFromWkt("POLYGON((0 0,0 1,1 1,0 0),())")
ERROR 5: OGR Error: Corrupt data
>>> g = ogr.CreateGeometryFromWkt("POLYGON((0 0,0 1,1 1,0 0),EMPTY)")
>>> g.ExportToIsoWkt()
'POLYGON ((0 0,0 1,1 1,0 0))'
>>> g.ExportToIsoWkb()
bytearray(b'\x01\x03\x00\x00\x00\x02\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
>>> ''.join('%02X' % b for b in g.ExportToIsoWkb())
'01030000000200000004000000000000000000000000000000000000000000000000000000000000000000F03F000000000000F03F000000000000F03F0000000000000000000000000000000000000000'
```

PostGIS:
```
$ psql -d autotest
psql (12.20 (Ubuntu 12.20-0ubuntu0.20.04.1))
Type "help" for help.

autotest=# select st_geomfromtext('POLYGON((0 0,0 1,1 1,0 0),())');
ERROR:  parse error - invalid geometry
HINT:  "POLYGON((0 0,0 1,1 1,0 0),())" <-- parse error at position 37 within geometry
autotest=# select st_geomfromtext('POLYGON((0 0,0 1,1 1,0 0),EMPTY)');
ERROR:  parse error - invalid geometry
HINT:  "POLYGON((0 0,0 1,1 1,0 0),EMPTY)" <-- parse error at position 40 within geometry
autotest=# select st_astext('01030000000200000004000000000000000000000000000000000000000000000000000000000000000000F03F000000000000F03F000000000000F03F0000000000000000000000000000000000000000'::geometry);
           st_astext           
-------------------------------
 POLYGON((0 0,0 1,1 1,0 0),())
(1 row)
```

So no interoperability on this subject, and neither GDAL or PostGIS manage to round-trip such geometries.
I would suggest QGIS to remove empty inner rings from its WKT export.

Reply:
> would suggest QGIS to remove empty inner rings from its WKT export.

+1

Reply:
Done.